refactor: Simplify code

aragon999 opened the pull request:
### 1. Why is this change necessary?
These are some changes which accumulated over time, when I was going through the code.

### 2. What does this change do, exactly?
It contains mostly cleanups and simplifications, which only should improve the developer experience and make use of new PHP features.

Not sure if such a PR is appreciated or not, or if it should be split into more smaller PRs.
Also I am not sure if a changelog is required for these changes.

### 3. Describe each step to reproduce the issue or behaviour.
Look at the code.

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

aragon999 committed:
refactor: Simplify and cleanup code

mitelg commented on src/Administration/Controller/UserConfigController.php:
```diff
@@ -73,7 +81,7 @@ private function getOwnUserConfig(Context $context, array $keys): array
         }
 
-        return $this->userConfigRepository->search($criteria, $context)->getElements();
+        return $this->userConfigRepository->search($criteria, $context)->getEntities()->getElements();
```
instead of returning the array, I would prefer returning the collection

mitelg commented on src/Core/Content/Product/Cart/ProductFeatureBuilder.php:
```diff
@@ -21,26 +22,36 @@ class ProductFeatureBuilder
     }
 
+    /**
+     * @param LineItem[] $lineItems
```
```suggestion
     * @param iterable<LineItem> $lineItems
```

mitelg commented on src/Core/Content/Product/Cart/ProductFeatureBuilder.php:
```diff
@@ -21,26 +22,36 @@ class ProductFeatureBuilder
     }
 
+    /**
+     * @param LineItem[] $lineItems
```
```suggestion
     * @param iterable<LineItem> $lineItems
```

mitelg commented on src/Core/Content/Product/Cart/ProductFeatureBuilder.php:
```diff
@@ -94,23 +101,25 @@ private function buildFeatures(CartDataCollection $data, LineItem $lineItem, Sal
     }
 
+    /**
+     * @param LineItem[] $lineItems
```
```suggestion
     * @param iterable<LineItem> $lineItems
```

mitelg commented on src/Core/Content/Product/Cart/ProductFeatureBuilder.php:
```diff
@@ -162,6 +170,9 @@ private function isRequiredCustomField(string $name, SalesChannelProductEntity $
     }
 
+    /**
+     * @return array{label: string, value: array<mixed>, type: string}
```
is value really an array :thinking: 

```suggestion
     * @return array{label: string, value: mixed, type: string}
```
I guess? :thinking:

aragon999 replied:
Yup, missed that :see_no_evil:

mitelg commented on src/Core/Content/Product/Cart/ProductFeatureBuilder.php:
```diff
@@ -240,9 +256,10 @@ private function getCustomField(string $name, CartDataCollection $data, SalesCha
 
         $customField = $data->get($fieldKey);
-        $label = $this->getCustomFieldLabel($customField);
+        \assert($customField instanceof CustomFieldEntity);
```
maybe do a real if condition instead :thinking: or can we really be sure, this is always true?

aragon999 replied:
Mh, so the data key is `custom-field-{$customFieldName}` and it has been checked that is exists. So it actually should always be a `CustomFieldEntity` otherwise something is broken.

mitelg replied:
if condition that then throws an exception maybe?

aragon999 replied:
Mh, okay :-D I changed the behavior to throw an exception.

mitelg commented on src/Core/Content/Test/ProductExport/Service/ProductExporterTest.php:
```diff
@@ -58,36 +56,12 @@ public function testExport(): void
-        static::assertCount(4, $csvRows);
-    }
-
-    public function testExportPagination(): void
```
why did you remove this test?

aragon999 replied:
As it is actually the same as the `testExport()` when the "useles" assert is removed :-)

mitelg reviewed (changes requested):
really nice! thank you very much for that :+1: I like such work :grin:

aragon999 commented:
@mitelg Thank you :-) Yes from time to time we need to cleanup :-D 

I adjusted the obvious changes, and added comments to the other ones. Additionally I also had some changes for the `ShippingMethodValidator` and the `CleanPersonalDataCommand`. I added them in a new commit (but I can also fixup them, if desired).

mitelg commented:
now some tests are failing, because of the changed exception messages :wink:

aragon999 commented:
Let's wait for the pipeline, but I hope now the tests are fixed (there was actually a bug in my changes :-) )

ghost commented:
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-37342

Please use this issue to track the state of your pull request.

cyl3x commented:
@aragon999 Thanks for your contribution! 🎉
